env0/resource_project_test.go

```diff
@@ -51,7 +51,11 @@ func TestUnitProjectResource(t *testing.T) {
 			Description: updatedProject.Description,
 		}).Times(1).Return(updatedProject, nil)
 
-		mock.EXPECT().Project(project.Id).AnyTimes()
+		gomock.InOrder(
+			mock.EXPECT().Project(gomock.Any()).Times(2).Return(project, nil),        // 1 after create, 1 before update
```

Comment:
this looks like you made the tests more specific but does it test the changes you made here?

Reply:
Yes, it's not only that is more specific. 
This flows calls `read` 3 times: 1 after create, 1 before update and 1 after update.
In case any of the read operations results in a plan with changes, the test fails so this is why previous mock did not work and I had to change it